core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1174,23 +1174,39 @@ object LogManager {
             time: Time,
             brokerTopicStats: BrokerTopicStats,
             logDirFailureChannel: LogDirFailureChannel): LogManager = {
+    this(config, initialOfflineDirs, Some(zkClient), brokerState, kafkaScheduler, time, brokerTopicStats, logDirFailureChannel)
+  }
+
+  def apply(config: KafkaConfig,
```

Comment:
Looking at the `LogManager` class more, I don't think it should contain `topicConfigs` at all.  That map doesn't get updated when a dynamic configuration changes.  It duplicates the configurations found in the `Log` objects themselves, except that it's not up to date (it's wrong, except maybe on startup).

We don't have time to change this now, but I'll file a JIRA